torch/utils/checkpoint.py

```diff
@@ -104,16 +104,18 @@ def backward(ctx, *args):
         # run backward() with only tensor that requires grad
         outputs_with_grad = []
         args_with_grad = []
+        args_index = 0
         for i in range(len(outputs)):
-            if outputs[i].requires_grad:
+            if torch.is_tensor(outputs[i]) and outputs[i].requires_grad:
                 outputs_with_grad.append(outputs[i])
-                args_with_grad.append(args[i])
+                args_with_grad.append(args[args_index])
+                args_index += 1
         if len(outputs_with_grad) == 0:
             raise RuntimeError(
                 "none of output has requires_grad=True,"
                 " this checkpoint() is not necessary")
         torch.autograd.backward(outputs_with_grad, args_with_grad)
-        grads = tuple(inp.grad if isinstance(inp, torch.Tensor) else inp
+        grads = tuple(inp.grad if isinstance(inp, torch.Tensor) else None
```

Comment:
Is there a reason we originally returned `inp` for grad if `inp` is not a Tensor? Why does it need to be `None` now? Just curious